test/jdk/javax/swing/text/html/bug8314731.java

```diff
@@ -0,0 +1,126 @@
+import java.awt.Container;
+import java.awt.Dimension;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JButton;
+import javax.swing.JEditorPane;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.SwingUtilities;
+import javax.swing.text.Document;
+import javax.swing.text.html.HTMLEditorKit;
+import javax.swing.text.html.StyleSheet;
+
+public class bug8314731 {
+
+    private static boolean testPassed;
+    private static JFrame jf;
+    private static JEditorPane jEditorPane;
+
+    public static void main(String[] args) throws Exception {
+        new bug8314731();
+        if (testPassed) {
+            System.out.println("ok");
+        } else {
+            throw new RuntimeException("FormView doesn't support the alt attribute, see JDK-8314731.");
+        }
+    }
+
+    public bug8314731() throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(bug8314731::createAndSetVisibleUI);
+            testPassed = ContainsAlt(jEditorPane);
+        } finally {
+            SwingUtilities.invokeAndWait(new Runnable() {
+                public void run() {
+                    jf.dispose();
+                }
+            });
+        }
+    }
+
+    private static void createAndSetVisibleUI() {
+
+        jEditorPane = new JEditorPane();
+        jEditorPane.setEditable(false);
+        jf = new JFrame("alt attribute test in HTML image type input");
+
+        JScrollPane scrollPane = new JScrollPane(jEditorPane);
+        HTMLEditorKit kit = new HTMLEditorKit();
+        jEditorPane.setEditorKit(kit);
+        StyleSheet styleSheet = kit.getStyleSheet();
+        styleSheet.addRule("""
+                body {
+                    color: #000;
+                    font-family:times;
+                    margin: 4px;
+                }
+                """);
+        String htmlString = """
+                <html>
+                    <body>
+                        <input type=image
+                               name=point
+                               alt="Logo">
+                    </body>
+                </html>
+                """;
+        Document doc = kit.createDefaultDocument();
+        jEditorPane.setDocument(doc);
+        jEditorPane.setText(htmlString);
+
+        jf.getContentPane().add(scrollPane, BorderLayout.CENTER);
+        jf.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        jf.setSize(new Dimension(400, 200));
+        jf.setLocationRelativeTo(null);
+        jf.setVisible(true);
+    }
+
+    private boolean ContainsAlt(Container container) {
+        for (Component c : container.getComponents()) {
+            if (c instanceof JButton butt) {
+                String text = butt.getText();
+                if (text.equals("Logo")) {
+                    return true;
+                }
+            } else if (c instanceof Container cont) {
+                if (ContainsAlt(cont)) {
+                    return true;
+                }
```

Comment:
```suggestion
                return ContainsAlt(cont));
```
There's no need for the `if`-statement here, too.